test/evm/foundry/local/Arbitrum_L3_Adapter.t.sol

```diff
@@ -0,0 +1,141 @@
+
+    receive() external payable {}
+}
+
+contract Token_ERC20 is ERC20 {
+    constructor(string memory name, string memory symbol) ERC20(name, symbol) {}
+
+    function mint(address to, uint256 value) public virtual {
+        _mint(to, value);
+    }
+
+    function burn(address from, uint256 value) public virtual {
+        _burn(from, value);
+    }
+}
+
+contract MinimalWeth is Token_ERC20 {
+    constructor(string memory name, string memory symbol) Token_ERC20(name, symbol) {}
+
+    function withdraw(uint256 amount) public {
+        _burn(msg.sender, amount);
+        (bool success, ) = payable(msg.sender).call{ value: amount }("");
+        require(success);
+    }
+}
+
+contract CrossDomainMessenger {
+    event MessageSent(address indexed target);
+
+    function sendMessage(
+        address target,
+        bytes calldata,
+        uint32
+    ) external {
+        emit MessageSent(target);
+    }
+}
+
+contract StandardBridge {
+    event ETHDepositInitiated(address indexed to, uint256 amount);
+
+    function depositERC20To(
+        address l1Token,
+        address l2Token,
+        address to,
+        uint256 amount,
+        uint32,
+        bytes calldata
+    ) external {
+        Token_ERC20(l1Token).burn(msg.sender, amount);
+        Token_ERC20(l2Token).mint(to, amount);
+    }
+
+    function depositETHTo(
+        address to,
+        uint32,
+        bytes calldata
+    ) external payable {
+        emit ETHDepositInitiated(to, msg.value);
+    }
+}
+
+contract ArbitrumL3AdapterTest is Test {
+    Rerouter_Adapter l3Adapter;
+    Optimism_Adapter optimismAdapter;
+
+    Token_ERC20 l1Token;
+    Token_ERC20 l2Token;
+    Token_ERC20 l1Weth;
+    Token_ERC20 l2Weth;
+    CrossDomainMessenger crossDomainMessenger;
+    StandardBridge standardBridge;
+
+    address l2Target;
+
+    function setUp() public {
+        l2Target = makeAddr("l2Target");
+
+        l1Token = new Token_ERC20("l1Token", "l1Token");
+        l2Token = new Token_ERC20("l2Token", "l2Token");
+        l1Weth = new MinimalWeth("l1Weth", "l1Weth");
+        l2Weth = new MinimalWeth("l2Weth", "l2Weth");
+
+        crossDomainMessenger = new CrossDomainMessenger();
+        standardBridge = new StandardBridge();
+
+        optimismAdapter = new Optimism_Adapter(
+            WETH9Interface(address(l1Weth)),
+            address(crossDomainMessenger),
+            IL1StandardBridge(address(standardBridge)),
+            IERC20(address(0)),
+            ITokenMessenger(address(0))
+        );
+        l3Adapter = new Mock_Rerouter_Adapter(address(optimismAdapter), l2Target);
+    }
+
+    // Messages should be indiscriminately sent to the l2Forwarder.
+    function testRelayMessage(address target, bytes memory message) public {
+        vm.expectEmit(address(crossDomainMessenger));
+        emit CrossDomainMessenger.MessageSent(l2Target);
+        l3Adapter.relayMessage(target, message);
```

Comment:
can you assert l2Target != target to make sure no false positive?

Reply:
Added vm.assume to enforce that they aren't equal. Then the expectEmit should do the checking that the correct argument was passed into the bridge internally. https://github.com/across-protocol/contracts/pull/607/commits/6d8a6339a81368fc47271c89581408d5d8b94f1b